daffodil-core/src/main/scala/org/apache/daffodil/core/runtime1/SchemaSetRuntime1Mixin.scala

```diff
@@ -68,7 +68,10 @@ trait SchemaSetRuntime1Mixin {
   }.value
 
   def onPath(xpath: String): DFDL.DataProcessor = {
-    Assert.usage(!isError)
+    Assert.argCheck(
+      !isError,
+      s"Must check isError before calling onPath. ${getDiagnostics.mkString}",
+    )
```

Comment:
Had a thought, these files are in daffodil-core and so I kind of consider them as private API. So maybe they still want to call usage()? And then the onPath/parse/unparse/etc functions in the daffodil-sapi/japi use a public usage variant that throws an IllegalStateException?

Or maybe these functions are so close to the public API that they are still considered public?

Reply:
It would not hurt to call checkNotError() here. But since this is internal code, other usage checks like that args are non-null should just be unadorned `usage(test)` only with no other clutter AFAIC. An important goal for this Assert.usage(...) and similar is no code clutter. If you even put a message string in there that's already too much.